src/react/useQueries.ts

```diff
@@ -41,13 +55,29 @@ export function useQueries(queries: UseQueryOptions[]): UseQueryResult[] {
       mountedRef.current = false
       unsubscribe()
     }
-  }, [observer])
+  }, [someIsLoading])
 
-  React.useEffect(() => {
-    // Do not notify on updates because of changes in the options because
-    // these changes should already be reflected in the optimistic result.
-    observer.setQueries(defaultedQueries, { listeners: false })
-  }, [defaultedQueries, observer])
+  const handleReset = React.useCallback(() => {
+    errorResetBoundary.clearReset()
```

Comment:
I don't think we can just call this _during_ render, as this has a side-effect. In `useBaseQuery.ts`, we call this in the `catch` of the `throw observer.fetchOptimistic`.